tests/functional/schema/test_custom_schema.py

```diff
@@ -0,0 +1,219 @@
+        )
+        check_relations_equal(
+            adapter=project.adapter,
+            relation_names=("agg", f"{_TABLE_THREE_SCHEMA}_{project.test_schema}_macro.table_3"),
+        )
+
+
+class TestCustomSchemaWithPrefixAndDispatch(BaseTestCustomSchema):
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "custom_macro.sql": _CUSTOM_MACRO_W_CONFIG,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "models": {"schema": _CUSTOM_SCHEMA},
+            "dispatch": [
+                {
+                    "macro_namespace": "dbt",
+                    "search_order": ["test", "package_macro_overrides", "dbt"],
+                }
+            ],
+        }
+
+    def test__postgres__custom_schema_with_prefix_and_dispatch(
+        self, project, macros, project_config_update
+    ):
+        project.run_sql(_VALIDATION_SQL)
+        run_dbt(["deps"])
+        run_dbt(["seed"])
+        results = run_dbt(["run"])
+        assert len(results) == 3
+        table_results = {r.node.name: r.node.schema for r in results.results}
+        assert table_results["view_1"] == f"{_CUSTOM_SCHEMA}_{project.test_schema}_macro"
+        assert table_results["view_2"] == f"{_TABLE_TWO_SCHEMA}_{project.test_schema}_macro"
+        assert table_results["table_3"] == f"{_TABLE_THREE_SCHEMA}_{project.test_schema}_macro"
+        check_relations_equal(
+            adapter=project.adapter,
+            relation_names=("seed", f"{_CUSTOM_SCHEMA}_{project.test_schema}_macro.view_1"),
+        )
+        check_relations_equal(
+            adapter=project.adapter,
+            relation_names=("seed", f"{_TABLE_TWO_SCHEMA}_{project.test_schema}_macro.view_2"),
+        )
+        check_relations_equal(
+            adapter=project.adapter,
+            relation_names=("agg", f"{_TABLE_THREE_SCHEMA}_{project.test_schema}_macro.table_3"),
+        )
+
+
+class TestCustomSchemaWithCustomMacroFromModelName(BaseTestCustomSchema):
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "custom_macro.sql": _CUSTOM_MACRO_MULTI_SCHEMA,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "models": {"schema": _CUSTOM_SCHEMA},
+            "seeds": {
+                "quote_columns": False,
+            },
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            f"{_TABLE_ONE_DOT_MODEL_NAME}.sql": _TABLE_ONE,
+            f"{_TABLE_TWO_DOT_MODEL_NAME}.sql": _TABLE_TWO_DOT_MODEL,
+            "table_3.sql": _TABLE_THREE_DOT_MODEL,
+        }
+
+    def test__postgres__custom_schema_from_model_name(
+        self, project, macros, project_config_update
+    ):
+        project.run_sql(_VALIDATION_SQL)
+        run_dbt(["seed"])
+        # run_dbt(["build"])
```

Comment:
Any reason this was left in as a comment?

Reply:
mostly cause I forgot to delete it